rmw_connext_cpp/src/functions.cpp

```diff
@@ -297,36 +346,21 @@ rmw_create_publisher(
     goto fail;
   }
 
-  // allocates memory, but doesn't have to be freed.
-  // partition operater takes ownership of it.
-  name_tokens = rcutils_split_last(topic_name, '/');
-  if (name_tokens.size == 1) {
-    partition_str = DDS_String_dup(ros_topics_prefix);
-    topic_str = DDS_String_dup(name_tokens.data[0]);
-  } else if (name_tokens.size == 2) {
-    // TODO(Karsten1987): Fix utility function for this
-    size_t partition_length = strlen(ros_topics_prefix) + strlen(name_tokens.data[0]) + 2;
-    char * concat_str = reinterpret_cast<char *>(rmw_allocate(partition_length * sizeof(char)));
-    snprintf(concat_str, partition_length, "%s/%s", ros_topics_prefix, name_tokens.data[0]);
-    // Connext will call deallocate on this, passing ownership to connext
-    partition_str = DDS_String_dup(concat_str);
-    topic_str = DDS_String_dup(name_tokens.data[1]);
-    // free temporary memory
-    rmw_free(concat_str);
-  } else {
-    RMW_SET_ERROR_MSG("Illformated topic name");
+  if (!_process_topic_name(
+      topic_name,
+      qos_profile->avoid_ros_namespace_conventions,
+      &topic_str,
```

Comment:
@wjwwood When is the allocated memory of this variable being freed? Same for `partition_str` next line.

Reply:
Connext takes ownership of it according to this comment: https://github.com/ros2/rmw_connext/commit/69bc3e61b919ce5e88348456473917e12ead86dc#diff-becdf01440f1757df19f4775b534e066R251

I believe if you free it then connext later does and you get a double free.

Reply:
Currently it shows up as being leaked in `valgrind` with the `demo_nodes_cpp` `talker` / `listener`.

Reply:
You can try freeing it, but I'm pretty sure we were getting a double free before.

Reply:
Also, this is the wrong pull request to ask on I think because it was only moved into a function in this pr. The removed code already had this ownership comment in place.

Reply:
for reference, seems related to https://github.com/ros2/rcl/pull/161/files, maybe doing one will impact the other

Reply:
> for reference, seems related to https://github.com/ros2/rcl/pull/161/files, maybe doing one will impact the other

Those are not related. `topic_str` is a local variable within `rmw_connext_cpp`.

Reply:
Moved: https://github.com/ros2/rmw_connext/pull/221#discussion_r136465335